File: 0001-two-sum/0001-two-sum.c
int* twoSum(int* nums, int numsSize, int target, int* returnSize) { 
    static int result[2]; // static so it can be returned 
    *returnSize = 2; 
 
    for (int i = 0; i < numsSize - 1; i++) { 
        for (int j = i + 1; j < numsSize; j++) { 
            if (nums[i] + nums[j] == target) { 
                result[0] = i; 
                result[1] = j; 
                return result; 
            } 
        } 
    } 
    return NULL; 
}

File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Map to store number and its index
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            // If complement exists in map, return the pair of indices
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            // Store current number with its index
            map.put(nums[i], i);
        }

        // As per problem, there is always exactly one solution
        return new int[0]; 
    }
}
